### DISCUSSION TITLE
"Bridge" between DDEV & setting up projects in coolify? (yaml configs?)

### DISCUSSION THREAD

**Discussion by mandrasch** [coollabsio/coolify#2047] [Feature Requests]
Hey everyone,
 
I use [DDEV](https://ddev.com/) for local development of PHP projects.

In each project there is already a [`.ddev/config.yaml`](https://github.com/mandrasch/ddev-craftcms-vite/blob/main/.ddev/config.yaml) file which defines the PHP version as well as database type and version:

```
name: ddev-craftcms-vite
type: craftcms
docroot: web
php_version: "8.1"
webserver_type: nginx-fpm
database:
    type: mysql
    version: "8.0"
```

When I set up a new project / resource in coolify via GitHub, it would be awesome if these values could somehow be used to configure the new resources in coolify? 🪄 🧙‍♀️

I already saw [coolify templates](https://github.com/inluxc/coolify-community-templates), but they work like 1-click fresh installs - not for adding existing git repository projects if I understood it correctly.

So if anyone has an idea on how to achieve a bridge between DDEV and coolify - please let me know! 🤗

My dream is a workflow like Vercel, but for PHP projects:

1. Select GitHub repository in coolify
2. Config yaml file in repository is automagically parsed 🪄 🧙‍♀️
3. Project is ready with database and correct `.env` file values

Guess platform.sh works a bit that way with yaml config files (https://docs.platform.sh/add-services.html), but haven't used it yet. Would be awesome to store the coolify project config in git as well to make deployments easier?

(I did various experiments with deploying PHP / CraftCMS projects in the past, e.g. with ploi.io https://dev.to/mandrasch/install-craftcms-via-ploiio-on-hetzner-cloud-including-ddev-57f8 - currently working on a tutorial for coolify 🚀 )

Update: There is a new tutorial with Coolify and Docker Compose ✨ https://samuelreichor.at/blogs/craft-coolify

Thx in advance & cheers,
Matthias

**Comment by notflip** [comment]
Hey @mandrasch Any news on this?

**Reply by mandrasch** [reply]
Hey @notflip - would be cool, but no. 

But check out https://dev.to/mandrasch/simple-coolify-example-with-docker-compose-github-deployments-53m 

In the end of the article there is a link to full Craft CMS example

We could write a "DDEV config.yaml to Docker Compose translator/converter" for simple things like php version, etc.

**Reply by notflip** [reply]
Thanks @mandrasch I will, also this article (don't know if it's recent), seems interesting: https://samuelreichor.at/blogs/craft-coolify

**Reply by mandrasch** [reply]
@notflip yeah, it's the one I linked to in my article 👍 great work by @samuelreichor 👏

**Comment by rfay** [comment]
Minor typo in title, fixing will help readability. Brige -> Bridge

**Comment by takielias** [comment]
@mandrasch You can check https://github.com/takielias/ddev-coolify. It generates a Coolify-compatible Dockerfile from your DDEV local configuration.
